ambry-tools/src/main/java/com.github.ambry.tools/admin/ConcurrencyTestTool.java

```diff
@@ -0,0 +1,946 @@
+      ArgumentAcceptingOptionSpec<Integer> minBlobSizeInBytesOpt =
+          parser.accepts("minBlobSizeInBytes", "Minimum size of blobs to be uploaded")
+              .withOptionalArg()
+              .describedAs("minBlobSizeInBytes")
+              .ofType(Integer.class)
+              .defaultsTo(1);
+      ArgumentAcceptingOptionSpec<Integer> sleepTimeBetweenBatchPutsInMsOpt =
+          parser.accepts("sleepTimeBetweenBatchPutsInMs", "Time to sleep in ms between batch puts")
+              .withOptionalArg()
+              .describedAs("sleepTimeBetweenBatchPutsInMs")
+              .ofType(Integer.class)
+              .defaultsTo(50);
+      ArgumentAcceptingOptionSpec<Integer> sleepTimeBetweenBatchGetsInMsOpt =
+          parser.accepts("sleepTimeBetweenBatchGetsInMs", "Time to sleep in ms between batch Gets")
+              .withOptionalArg()
+              .describedAs("sleepTimeBetweenBatchGetsInMs")
+              .ofType(Integer.class)
+              .defaultsTo(100);
+      ArgumentAcceptingOptionSpec<String> hostNameOpt =
+          parser.accepts("hostName", "The hostname against which requests are to be made")
+              .withOptionalArg()
+              .describedAs("hostName")
+              .defaultsTo("localhost")
+              .ofType(String.class);
+      ArgumentAcceptingOptionSpec<Integer> portNumberOpt =
+          parser.accepts("port", "The port number to be used while contacing the host")
+              .withOptionalArg()
+              .describedAs("port")
+              .ofType(Integer.class)
+              .defaultsTo(6667);
+      ArgumentAcceptingOptionSpec<Boolean> deleteOnExitOpt =
+          parser.accepts("deleteOnExit", "Delete blobs on exiting the test")
+              .withOptionalArg()
+              .describedAs("deleteOnExit")
+              .ofType(Boolean.class)
+              .defaultsTo(true);
+      ArgumentAcceptingOptionSpec<Boolean> enableVerboseLoggingOpt =
+          parser.accepts("enableVerboseLogging", "Enables verbose logging if set to true")
+              .withOptionalArg()
+              .describedAs("enableVerboseLogging")
+              .ofType(Boolean.class)
+              .defaultsTo(false);
+
+      ArrayList<OptionSpec<?>> requiredArgs = new ArrayList<>();
+      requiredArgs.add(hardwareLayoutFilePath);
+      requiredArgs.add(partitionLayoutFilePath);
+
+      OptionSet options = parser.parse(args);
+      if (hasRequiredOptions(requiredArgs, options)) {
+        this.hardwareLayoutFilePath = options.valueOf(hardwareLayoutFilePath);
+        logger.trace("Hardware layout file path: {}", this.hardwareLayoutFilePath);
+        this.partitionLayoutFilePath = options.valueOf(partitionLayoutFilePath);
+        logger.trace("Partition layout file path: {}", this.partitionLayoutFilePath);
+      } else {
+        parser.printHelpOn(System.err);
+        throw new InstantiationException("Did not receive all required arguments for starting RestServer");
```

Comment:
Already the statement " System.err.println("Missing required argument " + opt);" in hasRequiredOptions() would have printed the missing argument